### ISSUE TITLE
firebase:shell dumps function definition object to console 

### ISSUE THREAD

**Issue by christophe-g** [firebase/firebase-tools#1525]
### [REQUIRED] Environment info
firebase-tools: 7.1.1
firebase-admin: 8.2.0
firebase-functions: 3.1.0
node: 8.15.0
Linux

### [REQUIRED] Test case

Testing a function in the shell console cause a pretty big Object looking like a definition of all functions to be dumped in the console. This seems to be only happening after a threshold (either number of functions or volume of code). 

console.info messages are mixed into this dump: 
![image](https://user-images.githubusercontent.com/1168053/61542990-e51da700-aa42-11e9-97e4-2caf1731921b.png)

The Object looks like: 
```json

{
  "timestamp": "Fri Jul 19 2019 16:03:28 GMT+0200 (CEST)",
  "level": "SYSTEM",
  "text": "",
  "data":
  {
    "triggers":
    {
      "api":
      {
        "definition":
        {
          "httpsTrigger":
          {},
          "name": "api",
          "entryPoint": "api"
        },
        "module":
        {
          "appState":
          {},
          "arithmetic":
          {
            "counter":
            {},
            "average":
            {}
          },
          "delete":
          {},
          "entity":
          {
            "resource":
            {},
            "tenant":
            {},
            "program":
            {},
            "record":
            {},
            "business":
            {}
          },
          "resources":
          {
            "machine":
            {},
            "category":
            {}
          },
          "lowerCase":
          {},
          "membership":
          {},
          "user":
          {},
          "documentation":
          {},
          "schedule":
          {}
        }
      },
      "appState-userUpdateProgram":
      {
        "definition":
        {
          "eventTrigger":
          {
            "resource": "projects/_/instances/project-8476440599955222702/refs/userData/appState/{uid}/buid",
            "eventType": "providers/google.firebase.database/eventTypes/ref.write",
            "service": "firebaseio.com"
          },
          "name": "appState-userUpdateProgram",
          "entryPoint": "appState.userUpdateProgram"
        },
        "module":
        {
....
```

**Comment by samtstern** [comment]
This feels like it could be related to my recent change here, but I don't see any logging that's out of place:
https://github.com/firebase/firebase-tools/pull/1487

**Comment by christophe-g** [comment]
@samtstern - thanks a lot for the rapid feedback.

"MAIN" and "MemberType member" in the original screenshot are console.info messages. Agree, it is not obvious !

**Comment by samtstern** [comment]
Could you give a full log output where this happens as a text file?  I have
a theory but I need to see the full log session to verify.

Sam

**Comment by christophe-g** [comment]
Pretty sure this is not what you ask for  :
[dump.txt](https://github.com/firebase/firebase-tools/files/3411682/dump.txt)

How do a get a log session ?

**Comment by samtstern** [comment]
@christophe-g that's actually fine!  Thank you.

**Comment by samtstern** [comment]
@christophe-g I have a speculative fix.

Can you try installing from my branch:
```
npm install -g git://github.com/firebase/firebase-tools.git#ss-ipc-logs --save
```

And then seeing if you can still reproduce this problem?

**Comment by christophe-g** [comment]
Thanks!

Almost
```
christophe ~/Programming/Polymer/preignition/functions $ sudo npm install -g git://github.com/firebase/firebase-tools.git#ss-ipc-logs --save
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/firebase-tools/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! path /usr/local/lib/node_modules/firebase-tools/lib/bin/firebase.js
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod
npm ERR! enoent ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/firebase-tools/lib/bin/firebase.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/christophe/.npm/_logs/2019-07-19T19_14_48_933Z-debug.log
```
[2019-07-19T18_53_26_739Z-debug.log](https://github.com/firebase/firebase-tools/files/3412415/2019-07-19T18_53_26_739Z-debug.log)


I can install other npm modules (like firebase-tools) without problem

**Comment by samtstern** [comment]
@christophe-g could you try it without the `-g` and without the `sudo`.  So just:
```
npm install --save git://github.com/firebase/firebase-tools.git#ss-ipc-logs
```

And then instead of running `firebase functions:shell` run `./node_modules/.bin/firebase functions:shell`

**Comment by christophe-g** [comment]
@samtstern 

Cloned and built locally. The console looks much better ; ) 
Fixed on my end !

**Comment by samtstern** [comment]
@christophe-g nice, thank you for testing!

**Comment by samtstern** [comment]
This fix has been released in `7.2.0`
